boards/BMS/Test/Src/Test_StateMachine.cpp

```diff
@@ -628,4 +628,120 @@ TEST_F(BmsStateMachineTest, check_state_transition_from_fault_to_init_with_air_n
+{
+    SetInitialState(App_GetInitState());
+
+    // In Discharge state, acceptible temp range is (-20, 60), should be unaffected by temp of 46 C
+    get_max_temp_degc_fake.return_val = 46.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_NE(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+
+    LetTimePass(state_machine, 1000);
+
+    get_max_temp_degc_fake.return_val = 61.0f;
+    LetTimePass(state_machine, 10);
+
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_overtemp_charge_state)
+{
+    SetInitialState(App_GetChargeState());
+
+    // In Charge state acceptible temp range is (0, 45)
+    get_max_temp_degc_fake.return_val = 46.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_undertemp_drive_state)
+{
+    SetInitialState(App_GetInitState());
+
+    // In Discharge state, acceptible temp range is (-20, 60), should be unaffected by temp of -1 C
+    get_min_temp_degc_fake.return_val = -1.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_NE(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+
+    LetTimePass(state_machine, 1000);
+
+    get_min_temp_degc_fake.return_val = -21.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_undertemp_charge_state)
+{
+    SetInitialState(App_GetChargeState());
+
+    // In Charge state acceptible temp range is (0, 45)
+    get_min_temp_degc_fake.return_val = -1.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_ts_discharge_overcurrent)
+{
+    SetInitialState(App_GetInitState());
+
+    // Max acceptable discharge current is 88.5A*3 = 265.5A
+    get_high_res_current_fake.return_val = 90.0f;
+    get_low_res_current_fake.return_val  = 90.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+TEST_F(BmsStateMachineTest, check_state_transition_to_fault_state_from_all_states_ts_charge_overcurrent)
+{
+    SetInitialState(App_GetChargeState());
+
+    // Max acceptable charge current is 23.6A * 3 = 70.8A
+    get_high_res_current_fake.return_val = 24.0f;
+    get_low_res_current_fake.return_val  = 24.0f;
+    LetTimePass(state_machine, 10);
+    ASSERT_EQ(App_GetFaultState(), App_SharedStateMachine_GetCurrentState(state_machine));
+}
+
+TEST_F(BmsStateMachineTest, check_precharge_fault)
```

Comment:
This test does not strictly test the state machine, could be moved into a different file

Reply:
I'm kinda wondering if some of these tests could be moved to a new fault test

Reply:
I definitely agree, especially for check_precharge_fault, however the other tests I wrote do test state machine behaviour. I think that given it's only one test maybe it should stay for the time being, but If I write more tests when moving the rest of the BMS faults out of the error table maybe I should separate it to its own test then. 

Thoughts? I just think it would be a little silly to have a dedicated test just for check_precharge_fault